nav2_costmap_2d/test/integration/test_collision_checker.cpp

```diff
@@ -264,6 +300,35 @@ class TestCollisionChecker : public nav2_util::LifecycleNode
     return costmap_msg;
   }
 
+  nav_msgs::msg::OccupancyGrid
+  toOccupancyGridMsg(nav2_costmap_2d::Costmap2D * costmap)
```

Comment:
Doesn't this code already exist in the costmap somewhere? Seems this is likely duplicate code.

Reply:
It is duplicate code, for the most part, coming from the `costmap_2d_publisher` class, with minor modifications. Would you prefer to create some utility function for this in the costmap library?
https://github.com/ros-planning/navigation2/blob/d2f2145a6100521bfee10d25caa415d77fa204bd/nav2_costmap_2d/src/costmap_2d_publisher.cpp#L107

Reply:
Yes. If practical. That's quite a lot of code to duplicate.

Reply:
I've created a `costmap_2d_utils.cpp` file to factor out these msg conversions